src/models/user_data_processing.ts

```diff
@@ -0,0 +1,217 @@
+}
+
+export const RetrievedUserDataProcessing = tag<IRetrievedUserDataProcessing>()(
+  t.intersection([
+    UserDataProcessing,
+    DocumentDbUtils.RetrievedDocument,
+    VersionedModel
+  ])
+);
+
+// tslint:disable-next-line: prettier
+export type RetrievedUserDataProcessing = t.TypeOf<typeof RetrievedUserDataProcessing>;
+
+function toRetrieved(
+  result: DocumentDb.RetrievedDocument
+): RetrievedUserDataProcessing {
+  return RetrievedUserDataProcessing.decode(result).getOrElseL(_ => {
+    throw new Error("Fatal, result is not a valid RetrievedUserDataProcessing");
+  });
+}
+
+function toBaseType(o: RetrievedUserDataProcessing): UserDataProcessing {
+  return pick(
+    ["userDataProcessingId", "fiscalCode", "choice", "status", "createdAt"],
+    o
+  );
+}
+
+function getModelId(o: UserDataProcessing): ModelId {
+  return userDataProcessingIdToModelId(
+    makeUserDataProcessingId(o.fiscalCode, o.choice)
+  );
+}
+
+export function makeUserDataProcessingId(
+  fiscalCode: FiscalCode,
+  choice: UserDataProcessingChoice
+): UserDataProcessingId {
+  return UserDataProcessingId.decode(`${fiscalCode}-${choice}`).getOrElseL(
+    () => {
+      throw new Error("Invalid User Data Processing id");
+    }
+  );
+}
+
+function updateModelId(
+  o: UserDataProcessing,
+  id: NonEmptyString,
+  version: NonNegativeNumber
+): NewUserDataProcessing {
+  return {
+    ...o,
+    id,
+    kind: "INewUserDataProcessing",
+    version
+  };
+}
+
+/**
+ * A model for handling UserDataProcessing
+ */
+export class UserDataProcessingModel extends DocumentDbModelVersioned<
+  UserDataProcessing,
+  NewUserDataProcessing,
+  RetrievedUserDataProcessing
+> {
+  /**
+   * Creates a new User Data Processing model
+   *
+   * @param dbClient the DocumentDB client
+   * @param collectionUrl the collection URL
+   */
+  constructor(
+    dbClient: DocumentDb.DocumentClient,
+    collectionUrl: DocumentDbUtils.IDocumentDbCollectionUri
+  ) {
+    super(
+      dbClient,
+      collectionUrl,
+      toBaseType,
+      toRetrieved,
+      getModelId,
+      updateModelId
+    );
+  }
+
+  /**
+   * Searches for one user data processing request to the provided id
+   *
+   * @param id
+   */
+  public findOneUserDataProcessingById(
+    id: UserDataProcessingId,
+    fiscalCode: FiscalCode
+  ): Promise<Either<DocumentDb.QueryError, Option<UserDataProcessing>>> {
+    return super.findLastVersionByModelId(
+      USER_DATA_PROCESSING_MODEL_ID_FIELD,
+      id,
+      USER_DATA_PROCESSING_MODEL_PK_FIELD,
+      fiscalCode
+    );
+  }
+
+  public async createOrUpdateByNewOne(
+    userDataProcessing: UserDataProcessing
+  ): Promise<Either<QueryError, UserDataProcessing>> {
+    const newId = makeUserDataProcessingId(
+      userDataProcessing.fiscalCode,
+      userDataProcessing.choice
+    );
+
+    const toUpdate: UserDataProcessing = {
+      userDataProcessingId: newId,
+      // tslint:disable-next-line: object-literal-sort-keys
```

Comment:
can we just sort the keys here ? =)

Reply:
Could be fine this way? -> 
`public async createOrUpdateByNewOne(
    userDataProcessing: UserDataProcessing
  ): Promise<Either<QueryError, UserDataProcessing>> {
    const newId = makeUserDataProcessingId(
      userDataProcessing.choice,
      userDataProcessing.fiscalCode
    );

    const toUpdate: UserDataProcessing = {
      choice: userDataProcessing.choice,
      createdAt: userDataProcessing.createdAt,
      fiscalCode: userDataProcessing.fiscalCode,
      status: userDataProcessing.status,
      updatedAt: userDataProcessing.createdAt,
      userDataProcessingId: newId
    };`